### PR TITLE
Add Service to Fortio so that the sidecar can become ready.

### PR DESCRIPTION
**Pull request by esnible**
For https://github.com/istio/istio/issues/12333

Uses the solution from https://github.com/istio/istio/issues/9504#issuecomment-439432130

### CODE REVIEW

**Comment by esnible** [comment]
Added statsInclusionPrefixes to resolve https://github.com/istio/istio.io/issues/3562

**Comment by esnible** [comment]
Adding this to 1.1 milestone because the Issues were added to 1.1 milestone.

File: samples/httpbin/sample-client/fortio-deploy.yaml [hunk 1]
```diff
@@ -6,6 +6,14 @@ spec:
   replicas: 1
   template:
     metadata:
+      annotations:
+        # This annotation allows istio-proxy sidecar to become ready.  Without
+        # it Envoy logs "Envoy proxy is NOT ready" and "envoy missing listener for inbound application port"
+        status.sidecar.istio.io/port: "0"
```
Comment (frankbu) [inline samples/httpbin/sample-client/fortio-deploy.yaml]: You should document better what's going on here. Disabling readiness? Why is this needed for Fortio, but not other clients, like the sleep sample?
^^^ @ldemailly

Comment (esnible) [reply samples/httpbin/sample-client/fortio-deploy.yaml]: @nmittler I did `port:0` because you recommended it on 
https://github.com/istio/istio/issues/9504#issuecomment-439432130 .  I don't understand why it is needed.  Can you explain it to Frank and I?

Comment (esnible) [reply samples/httpbin/sample-client/fortio-deploy.yaml]: @frankbu The sleep sample doesn't need this because it comes with a Service even though you might think it is headless.  _samples/sleep/sleep.yaml_.  I also wonder why Sleep is considered 2/2 Ready because it's Service is on port 80 but `netstat -tulpn` and `curl localhost:80` show it is not listening on that port.

Comment (frankbu) [reply samples/httpbin/sample-client/fortio-deploy.yaml]: Assuming this is a more general problem, I think we should add an entry in the operations guide to explain this issue. It can point to this Fortio deployment in the httpbin sample as an example. 

Maybe something about when you need to use the port 0 annotation in here: https://preliminary.istio.io/help/ops/traffic-management/deploy-guidelines/

The statsInclusion annotation should be explained somewhere too. Maybe here? https://preliminary.istio.io/help/ops/telemetry/

Comment (esnible) [reply samples/httpbin/sample-client/fortio-deploy.yaml]: I will write something about the `statsInclusionPrefixes` annotation.

**Comment by frankbu** [comment]
Given that adding the service, fixes the readiness problem, let's merge this, but I think we need to also address the problem when no service is defined. Should open another issue: If there are no inbound ports, then it should not perform a check.

File: samples/httpbin/sample-client/fortio-deploy.yaml [hunk 2]
```diff
@@ -6,6 +19,11 @@ spec:
   replicas: 1
   template:
     metadata:
+      annotations:
+        # This annotation causes Envoy to serve cluster.outbound statistics via 15000/stats
+        # in addition to the stats normally served by Istio.  The Circuit Breaking example task
+        # gives an example of inspecting Envoy stats.
+        sidecar.istio.io/statsInclusionPrefixes: cluster.outbound,cluster_manager,listener_manager,http_mixer_filter,tcp_mixer_filter,server,cluster.xds-grpc
```
Comment (rshriram) [inline samples/httpbin/sample-client/fortio-deploy.yaml]: why this? this seems gratuitous

Comment (esnible) [reply samples/httpbin/sample-client/fortio-deploy.yaml]: PR https://github.com/istio/istio/pull/10184 turned off most Envoy stats collection.  `curl localhost:15000/stats` returns only a few things.  This was done to save memory.  Unfortunately it turns off all stats for normal outbound and listener which means that the Circuit Breaking instructions didn't work.  It also turns off collections about TLS problems on listeners!

I did https://github.com/istio/istio/pull/11398 which allows an annotation to turn collection back on.  By including `cluster.outbound` the Circuit Breaking instruction to do `kubectl exec -it $FORTIO_POD  -c istio-proxy  -- sh -c 'curl localhost:15000/stats' | grep httpbin | grep pending` works again.  I am writing up documentation this morning on how to use this annotation.

**Comment by linsun** [comment]
@utka @duderino @louiscryan @rshriram can we get this merged in 1.1?  This fixes the circuit breaking task in istio.io.
